App/Sagas/TextileSagas.ts

```diff
@@ -156,6 +154,31 @@ export function * viewThread ( action: ActionType<typeof UIActions.viewThreadReq
   yield call(NavigationService.navigate, 'ViewThread', { id: action.payload.threadId, name: action.payload.threadName })
 }
 
+export function * getUsername (contact: TextileTypes.Contact) {
+  try {
+    if (contact.username !== undefined) return
+    const uri = contact.id ? Config.TEXTILE_CAFE_URI + '/ipns/' + contact.id + '/username' : undefined
+    const response = yield call(fetch, uri)
```

Comment:
What/where is `fetch`?

Reply:
fetch is built in url request method

Reply:
Ah right, forgot that it's available globally.